crates/shared/src/price_estimation/quasimodo.rs

```diff
@@ -103,24 +98,60 @@ impl QuasimodoPriceEstimator {
                     BigInt::from(*pool.fee.numer()),
                     BigInt::from(*pool.fee.denom()),
                 )),
-                cost: CostModel {
-                    amount: U256::from(GAS_PER_UNISWAP) * gas_price,
-                    token: self.native_token,
-                },
+                cost: gas_model.uniswap_cost(),
                 mandatory: false,
             })
+            .chain(balancer_pools)
             .enumerate()
             .collect();
 
+        // Solver cannot handle 0 reserves so filter these pools out until that is fixed.
+        amms.retain(|_, v| match &v.parameters {
+            AmmParameters::ConstantProduct(p) => p.reserves.values().all(|r| !r.is_zero()),
+            AmmParameters::WeightedProduct(p) => p.reserves.values().all(|r| !r.balance.is_zero()),
+            AmmParameters::Stable(p) => p.reserves.values().all(|r| !r.is_zero()),
```

Comment:
For some reason I thought there was a `has_sufficient_reserves()` method for this already.

Reply:
Changed to use it.